Prokopishena.University.Personalized-Fitness-Coaching-Platform.Web/server.js

```diff
@@ -0,0 +1,1063 @@
+
+  try {
+    const trainingPlan = await TrainingPlan.create({
+      trainer_id: trainer_id,
+      client_id: client_id,
+      plan_duration: plan_duration,
+      trainings_per_week: trainings_per_week
+    });
+
+    res.status(201).json({ message: 'Training plan created successfully', plan_id: trainingPlan.plan_id });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Помилка при створенні плану тренувань' });
+  }
+});
+app.post('/training-sessions', async (req, res) => {
+  const { plan_id, training_sessions } = req.body;
+
+  try {
+    for (const session of training_sessions) {
+      const trainingSession = await TrainingSession.create({
+        plan_id: plan_id,
+        session_number: session.session_number
+      });
+
+      for (const exercise of session.exercises) {
+        await TrainingExercise.create({
+          session_id: trainingSession.session_id,
+          exercise_id: exercise.exercise_id,
+          repetitions: exercise.repetitions,
+        });
+      }
+    }
+
+    res.status(201).json({ message: 'Training sessions created successfully' });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Помилка при створенні тренувальних сесій' });
+  }
+});
+
+
+app.get('/training-sessions/:sessionId', async (req, res) => {
+  try {
+    const sessionId = req.params.sessionId;
+    const session = await TrainingSession.findByPk(sessionId);
+    res.json(session);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Помилка при отриманні сесії тренувань' });
+  }
+});
+
+app.post('/exercises', async (req, res) => {
+  try {
+    const { exercise_name, description } = req.body;
+    const newExercise = await Exercise.create({
+      exercise_name,
+      description
+    });
+    res.status(201).json(newExercise);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Помилка при створенні вправи' });
+  }
+});
+
+app.get('/exercises', async (req, res) => {
+  try {
+    const exercises = await Exercise.findAll({ attributes: ['exercise_id', 'exercise_name'] });
+    res.json(exercises);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: 'Помилка при отриманні списку вправ' });
+  }
+});
+
+// Static file handler for public resources
+app.use(express.static(path.join(__dirname, 'public')));
+
+app.use('/static', express.static(path.join(__dirname, 'static')));
+
+// Route for serving HTML page
+app.get('/', (req, res) => {
+  res.sendFile(path.join(__dirname, 'index.html'));
+});
+
+// Exercise route
+app.get('/exercises', trainingController.getExercises);
+
+// Create training plan route
+app.post('/training-plans', trainingController.createTrainingPlan);
+
+
+// Handle chat functionality
+
+// Server start
+app.listen(PORT, () => {
+  console.log(`Server is running on port ${PORT}`);
+});
+
+

```

Comment:
The file is oversatured with too much lines of code. It would be better if it's separated into few smaller files.